src/types.hpp

```diff
@@ -11,5 +11,14 @@ const int GRID_SIZE = 8;
 class Piece;
 using BoardState = array<array<Piece *, GRID_SIZE>, GRID_SIZE>;
 
+//categorising the pieces
+enum class PieceType{PAWN, KNIGHT, BISHOP, ROOK, QUEEN, KING};
+
 // represnting the team of a piece
-typedef enum { White, Black } Team;
+enum class Team { WHITE, BLACK };
+
+
+//check if cell coordinates are valid coordinates on the board
+auto isValidCell = [](int x, int y) {
+  return (x >= 0 && x < GRID_SIZE) && (y >= 0 && y < GRID_SIZE);
+};
```

Comment:
petition to rename `isValidCell` to `withinBoard` or `withinBounds` as valid cell target will vary with chess peice